### ISSUE TITLE
Component blinks after update value when using image as fill

### ISSUE THREAD

**Issue by mdaveiga** [kottenator/jquery-circle-progress#134]
The situation is easy to reproduce, you can create a components like this: 

`    circle.circleProgress({
        value: 1.0,
        animation: false,
        fill: {
            color: "#1a1f21", image: "../img/clock.png"
        }
    });
`
Then try to update the value with a timer. You'll se the image blinks after some seconds. It may has something to do with the fact that the image is being recreated everytime it updates. I've seen some posts about it, and they say you need to call it once, and then work with the draw.

**Comment by kottenator** [comment]
Yes, you're right, due to current implementation, your example will re-create `Image()` object on  _each redraw_.

You can avoid it by using:

```js
el.circleProgress({
  fill: {
    image: <Image instance> // instead of URL string
  }
});
```

**Comment by kottenator** [comment]
@mdaveiga, @andreiglingeanu - do you think it's worth it to implement `Image` "caching" when `typeof fill.image == 'string'`? Or is it enough to use `Image` instance instead of string as a workaround?

**Comment by andreiglingeanu** [comment]
@kottenator You may try to implement the cache as a simple key-value store, in memory, where the key is the actual `fill.image` (just when it is a string). I'm seeing this cache as some data structure that is not stored inside _each_ `CircleProgress` instance, but rather a global one, so that you can share same `Image` objects that map to same URLs. In fact, this `fill.image` may also be a base64 string and this way your persistent cache will grow overtime. 

That's just something to be concerned about, if you'll approach an implementation of that thing.

But, to be honest, I'd rather document the fact that you'll have to use `Image` objects in some edge cases :)

<hr>

By the way, how's the work on removing jquery as dependency, I saw you created a new repo for that purpose? I'm interested in having that and let me know if you'd like a hand of help ;)

**Comment by kottenator** [comment]
@andreiglingeanu - about no-jQuery version: I haven't worked on it for awhile. The thing that stopped me was - which animation library to use? Could you suggest something?

I need it to be simple, stable, with "easing" setting, and allow animate custom variables, not only CSS properties.

**Comment by andreiglingeanu** [comment]
@kottenator I'm also guessing that you [don't need a way to animate CSS](https://github.com/kottenator/jquery-circle-progress/blob/1.2.1/dist/circle-progress.js#L418-L431) at all, you just need a general purpose timing function. 

 I'd use that and probably provide a way to plug something else in (but that's totally optional and you'll have to force some kind of contract):

```javascript

/**
 * duration: duration in ms
 * easing: an easing function, below you'll have a couple of them provided
 * draw: render UI for current state
 */
function animate(options) {

  // docs: https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
  // compatibility: http://caniuse.com/#feat=high-resolution-time
  var start = performance.now();

  requestAnimationFrame(function animate(time) {
    // timeFraction from 0 to 1
    var timeFraction = (time - start) / options.duration;
    if (timeFraction > 1) timeFraction = 1;

    // текущее состояние анимации
    var progress = options.easing(timeFraction)

    options.draw(progress);

    if (timeFraction < 1) {
      requestAnimationFrame(animate);
    }

  });
}

// a couple of easings implemented to use
// more info about them in the article I'll link below (russian)
// playground for it: http://plnkr.co/edit/xPP9UTENxFSPnxCYQefq?p=preview
function linear (timeFraction) {
  return timeFraction;
}

function ease (timeFraction) {
  return 1 - Math.sin(Math.acos(timeFraction))
}

function makeEaseOut(timing) {
  return function(timeFraction) {
    return 1 - timing(1 - timeFraction);
  }
}

function makeEaseInOut(timing) {
  return function(timeFraction) {
    if (timeFraction < .5)
      return timing(2 * timeFraction) / 2;
    else
      return (2 - timing(2 * (1 - timeFraction))) / 2;
  }
}

var easeOut = makeEaseOut(ease);
var easeInOut = makeInEaseOut(ease);

```

Disclaimer: That's not my function. I've stolen it from [here](https://learn.javascript.ru/js-animation#итого). High quality content there, though in russian.
